File: agent/maestro_agent/services/maestro_api/agent_log.py
import dateutil.parser
from maestro_agent.services.maestro_api import MaestroApiClient


class AgentLog:
    def __init__(self, id, agent_id, log_message, created_at, updated_at):
        self.id = id
        self.agent_id = agent_id
        self.log_message = log_message
        self.created_at = created_at
        self.updated_at = updated_at


def agent_log_json_to_object(job_json):

    return AgentLog(
        id=job_json.get("id"),
        agent_id=job_json.get("agent_id"),
        log_message=job_json.get("log_message"),
        created_at=dateutil.parser.parse(job_json.get("created_at")),
        updated_at=dateutil.parser.parse(job_json.get("updated_at")),
    )


class AgentLogApi:
    @staticmethod
    def send_log_message(agent_id, log_message, level):

        data = dict(agent_id=agent_id, log_message=log_message, level=level)
        return MaestroApiClient.post(
            "/api/agent_log",
            data=data,
            mapper=agent_log_json_to_object,
        )

File: web/frontend/src/lib/api/endpoints/agentLog.js
import ErrorHandler from "../../../ErrorHandler";
import { toLocalDate, toUtcString } from "../../date";
import { maestroClient } from "../../services/maestroApi";
import { agentLogLevel } from "../models";

const agentLogObjectMapper = (agentLog) => ({
  id: agentLog.id,
  agentId: agentLog.agent_id,
  logMessage: agentLog.log_message,
  level: agentLogLevel[agentLog.level],
  createdAt: toLocalDate(agentLog.created_at),
  updatedAt: toLocalDate(agentLog.updated_at)
});

/**
 *
 * @param {Moment} dateFrom // agentLog created at should be biiger than
 * @param {Array} agentIds
 * @returns
 */
export const fetchAgentLogs = async ({
  dateFrom,
  levels = [],
  agentIds = []
}) => {
  try {
    const res = await maestroClient.get("/api/agent_logs", {
      params: {
        date_from: toUtcString(dateFrom),
        log_levels: levels,
        agent_ids: agentIds
      }
    });

    const agentLogs = res.data.map(agentLogObjectMapper);

    return agentLogs;
  } catch (error) {
    ErrorHandler.handleError(error, "agent logs");
    throw error;
  }
};

export const clearAgentLog = async (agentId) => {
  try {
    const res = await maestroClient.delete(`/api/agent_log/${agentId}`);

    const numDeleteLogs = res.data;

    return numDeleteLogs;
  } catch (error) {
    ErrorHandler.handleError(error, "agent logs");
    throw error;
  }
};
